File: 3005. Count Elements With Maximum Frequency.py
class Solution(object):
    def maxFrequencyElements(self, nums):
        dic={}
        mx=0
        count=0
        flag=0
        for i in range(len(nums)):
            if nums[i] in dic.keys():
                flag=1
                dic[nums[i]]+=1
                if dic[nums[i]]>mx:
                    mx=dic[nums[i]]
                    count=1
                elif dic[nums[i]]==mx:
                    count+=1
            else:
                dic[nums[i]]=1
        if flag==1:
            return mx*count
        else:
            return len(nums)

File: 3005. Count Elements With Maximum Frequency.java


class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        frequencies = {}
        max_ = 0
        for n in nums:
            frequencies[n] = frequencies.get(n, 0) + 1
            max_ = max(max_, frequencies[n])
        
        ans = 0
        for v in frequencies.values():
            if v == max_:
                ans += v
        
        return ans
